lib/core.ps1

```diff
@@ -661,14 +661,15 @@ function Invoke-ExternalCommand {
         } else {
             # escape arguments manually in lower versions, refer to https://docs.microsoft.com/en-us/previous-versions/17w5ykft(v=vs.85)
             $escapedArgs = $ArgumentList | ForEach-Object {
-                # escape N consecutive backslash(es), which are followed by a double quote, to 2N consecutive ones
-                $s = $_ -replace '(\\+)"', '$1$1"'
-                # escape N consecutive backslash(es), which are at the end of the string, to 2N consecutive ones
-                $s = $s -replace '(\\+)$', '$1$1'
-                # escape double quotes
-                $s = $s -replace '"', '\"'
-                # quote the argument
-                "`"$s`""
+                # escape N consecutive backslash(es), which are followed by a double quote or at the end of the string, to 2N consecutive ones
+                $s = $_ -replace '(\\+)(""|$)', '$1$1$2'
+                # quote the path if it contains spaces and is not NSIS's '/D' argument
+                # ref: https://nsis.sourceforge.io/Docs/Chapter3.html
+                if ($s -match ' ' -and $s -notmatch '/D=[A-Z]:[\\/].*') {
+                    $s = $s -replace '([A-Z]:[\\/].*)', '`"$1`"'
```

Comment:
This doesn't seem to work in PS5. Test manifest:

cygwin.json
```json
{
    "version": "2.931",
    "description": "A large collection of GNU and Open Source tools which provide functionality similar to a Linux distribution on Windows",
    "homepage": "https://cygwin.com/",
    "license": {
        "identifier": "GPL-3.0-or-later,X11,Public Domain",
        "url": "https://cygwin.com/faq.html#faq.what.free"
    },
    "notes": "To start a Cygwin shell, type \"cygwin\".",
    "architecture": {
        "64bit": {
            "url": "https://cygwin.com/setup-x86_64.exe#/cygwin-setup.exe",
            "hash": "9aeacdcf8ff71e2d1a1f9b4685eac12c4de4bb312d44ddac2c7b226f5952bfde"
        },
        "32bit": {
            "url": "https://cygwin.com/setup-x86.exe#/cygwin-setup.exe",
            "hash": "2f1fb00d2693c431d6878799fce00e86a1a238afb8fd4289590708861fb27d8a"
        }
    },
    "installer": {
        "args": [
            "--no-admin",
            "--no-shortcuts",
            "--quiet-mode",
            "--local-package-dir",
            "C:\\Tmp\\John Wick\\packages",
            "--root",
            "C:\\Tmp\\John Wick\\root",
            "--site",
            "https://mirrors.ustc.edu.cn/cygwin/"
        ],
        "keep": true
    },
    "bin": [
        [
            "root\\bin\\bash.exe",
            "cygwin",
            "--login -i"
        ]
    ],
    "shortcuts": [
        [
            "root\\Cygwin.bat",
            "Cygwin",
            "",
            "root\\Cygwin-Terminal.ico"
        ],
        [
            "cygwin-setup.exe",
            "Cygwin Setup",
            "--no-admin --no-shortcuts --root \"$persist_dir\\root\" --local-package-dir \"$persist_dir\\packages\""
        ],
        [
            "root\\bin\\mintty.exe",
            "Cygwin Terminal",
            "-i \"$dir\\root\\Cygwin-Terminal.ico\" -",
            "root\\Cygwin-Terminal.ico"
        ]
    ],
    "checkver": {
        "url": "https://cygwin.com/git/?p=cygwin-apps/setup.git;a=tags",
        "regex": ">release_([\\d.]+)<"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://cygwin.com/setup-x86_64.exe#/cygwin-setup.exe"
            },
            "32bit": {
                "url": "https://cygwin.com/setup-x86.exe#/cygwin-setup.exe"
            }
        }
    }
}
```

The path `C:\\Tmp\\John Wick\\` is a simulation.

Reply:
OK, in PS5 the assign statement doesn't return value, and I made a stupid mistake.